modules/tracktion_engine/playback/devices/tracktion_WaveInputDevice.cpp

```diff
@@ -1203,7 +1183,22 @@ class WaveInputDeviceInstance  : public InputDeviceInstance
                     }
                     else
                     {
-                        addBlockToRecord (inputBuffer, adjustSamples, numSamples - adjustSamples);
+                        // BEATCONNECT MODIFICATION START (RELAY)
+                        auto& wi = getWaveInput();
+                        if (wi.getIsRelay())
+                        {
+                            addBlockToRecord(inputBuffer, 0, numSamples);
```

Comment:
Shouldn't this be the following:
`addBlockToRecord(inputBuffer, adjustSamples, numSamples - adjustSamples);`

Reply:
For the relay, we have to consider the full size of the first input buffer. Or else we will have the first adjustSamples missing from the recorded clip.

Reply:
As for the video, do you mean that the progress of the recording should be 2 channels?